assets/js/blocks/stock-filter/block.js

```diff
@@ -243,12 +243,8 @@ const StockStatusFilterBlock = ( {
 	useEffect( () => {
 		if ( filteringForPhpTemplate ) {
 			setChecked( checked );
-			// Only automatically redirect if the filter button is not active.
-			if ( ! blockAttributes.showFilterButton ) {
-				redirectPageForPhpTemplate( checked );
-			}
```

Comment:
@tjcafferkey in my test, we only need the call of `redirectPageForPhpTemplate` inside the `onSubmit`. When the filter button is disabled, `onSubmit` is called every time the checked value changes.

Reply:
I get the same results in my tests too thanks for correcting this.